koku/masu/database/trino_sql/reporting_ocpazurecostlineitem_daily_summary.sql

```diff
@@ -285,7 +298,322 @@ GROUP BY coalesce(azure.date, azure.usagedatetime),
     azure.matched_tag
 ;
 
--- Directly resource_id matching
+{% if unattributed_storage %}
+INSERT INTO hive.{{schema | sqlsafe}}.azure_openshift_disk_capacities_temp (
+    resource_id,
+    capacity,
+    usage_start,
+    ocp_source,
+    year,
+    month
+)
+WITH cte_ocp_filtered_resources as (
+    SELECT
+        distinct azure.resource_id as azure_partial_resource_id
+    FROM hive.{{schema | sqlsafe}}.reporting_ocpusagelineitem_daily_summary as ocp
+    JOIN hive.{{schema | sqlsafe}}.azure_openshift_daily_resource_matched_temp as azure
+        ON (azure.usage_start = ocp.usage_start)
+        AND (
+                (strpos(azure.resource_id, ocp.persistentvolume) > 0 AND ocp.data_source = 'Storage')
+            OR
+                (lower(ocp.csi_volume_handle) = lower(azure.resource_id))
+            )
+    WHERE ocp.source = {{ocp_source_uuid}}
+        AND ocp.year = {{year}}
+        AND lpad(ocp.month, 2, '0') = {{month}}
+        AND ocp.usage_start >= {{start_date}}
+        AND ocp.usage_start < date_add('day', 1, {{end_date}})
+        AND azure.ocp_source = {{ocp_source_uuid}}
+        AND azure.year = {{year}}
+        AND azure.month = {{month}}
+)
+SELECT
+    ocp_filtered.azure_partial_resource_id,
+    max(az_disk_capacity.capacity) as capacity,
+    date(coalesce(date, usagedatetime)) as usage_start,
+    {{ocp_source_uuid}} as ocp_source,
+    {{year}} as year,
+    {{month}} as month
+FROM azure_line_items as azure
+JOIN postgres.public.reporting_common_diskcapacity as az_disk_capacity
+    ON azure.metername LIKE '%' || az_disk_capacity.product_substring || ' %' -- space here is important to avoid partial matching
+    AND az_disk_capacity.provider_type = 'Azure'
+JOIN cte_ocp_filtered_resources as ocp_filtered
+    ON split_part(coalesce(nullif(azure.resourceid, ''), azure.instanceid), '/', 9) = ocp_filtered.azure_partial_resource_id
+WHERE coalesce(azure.date, azure.usagedatetime) >= TIMESTAMP '{{start_date | sqlsafe}}'
```

Comment:
(issue, blocking) This WHERE clause needs the partitioning filters:
```
WHERE ...
        AND azure.year = {{year}}
        AND azure.month = {{month}}
```

Reply:
Conditions added! Great catch!